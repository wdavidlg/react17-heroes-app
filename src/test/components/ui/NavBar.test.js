const { mount } = require("enzyme")
const { MemoryRouter, Router } = require("react-router")
const { AuthContext } = require("../../../auth/AuthContext")
const { NavBar } = require("../../../componentes/ui/NavBar")

describe('Pruebas en <NavBar/>', () => {

    const historyMock = {
        replace: jest.fn(),
        listen: jest.fn(),
        location: {},
        createHref: jest.fn()
    }

    const contextValue = {
        user: {
            name: 'Will',
            logged: true
        },
        dispatch: jest.fn()
    }

    const wrapper = mount(
        <MemoryRouter>
            <AuthContext.Provider value={contextValue}>
                <Router history={historyMock}>
                    <NavBar />
                </Router>
            </AuthContext.Provider>
        </MemoryRouter>
    )

    afterEach(()=>{
        jest.clearAllMocks();
    })

    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span.text-info').text()).toBe('Will')
    })

    test('debe de llamar al dispatch y history.replace', () => {
        wrapper.find('button.btn').simulate('click');
        expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
        expect(historyMock.replace).toHaveBeenCalledTimes(1);
        expect(historyMock.replace).toHaveBeenLastCalledWith('/login')
    })
    

})
