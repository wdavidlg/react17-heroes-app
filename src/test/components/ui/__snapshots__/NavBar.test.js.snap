// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavBar/> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNavAltMarkup"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <a
          aria-current={null}
          className="nav-link"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-link"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-link"
          onClick={[Function]}
        >
          Search
        </a>
        <button
          className="nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
      <span
        className="navbar-text text-info"
      >
        Will
      </span>
    </div>
  </div>
</nav>
`;
